tests/test_purge.py

```diff
@@ -131,20 +132,21 @@ def test_scheme_port(client_auth, requests_mocker):
     assert future.result()
 
 
-def test_response_fails(client, requests_mocker, no_retries):
+@responses.activate
+def test_response_fails(client,  no_thread_retries):
     """Requests fail with a FastPurgeError if API gives unsuccessful response."""
+    url = 'https://fastpurge.example.com/ccu/v3/delete/cpcode/production'
+    # Decrease backoff, otherwise the test will run for 5 minutes
+    with pytest.MonkeyPatch.context() as mp:
```

Comment:
Note that you can just declare a 'monkeypatch' argument in the test function:

```
def test_response_fails(client,  no_thread_retries, monkeypatch):
    monkeypatch.setenv(...)
```

...then you don't need to manage the context yourself, which avoids having to nest all that code.